compiler/src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -454,6 +454,37 @@ object Contexts {
     def freshOver(outer: Context): FreshContext =
       FreshContext(base).init(outer, this).setTyperState(this.typerState)
 
+    /** Crash the compiler with a `Throwable` in a controlled way, reporting useful info
+     * and asking users to submit a crash report.
+     * This helps users by providing information they can use to work around the crash
+     * and helps compiler maintainers by capturing important information about the crash.
+     * With the exception of `ControlThrowable`s, this method should be used instead of 
+     * throwing exceptions whenever a context is available.
+     * 
+     * instead of:
+     * `throw Exception("foo")` 
+     * use:
+     * `ctx.implode(Exception("foo"))`
+     */
+    inline def implode(inline cause: Throwable): Nothing = 
```

Comment:
```suggestion
    def implode(inline cause: Throwable): Nothing = 
```

Reply:
We will not gain anything from inlining this definition. We might actually end up generating more code.